42/README.md

```diff
@@ -0,0 +1,71 @@
+```
+shortname: [42/HARD-FORKS]
+name: Handling Hard Forks
+type: information
+Status: raw
+editor: Vanshdeep Singh <vanshdeep@bigchaindb.com>
+```
+
+## Description
+
+Currently, there is no active strategy to address breaking changes in BigchainDB 2.0 i.e. there is no general approach to migration such that breaking changes in transaction specs can be handled. This document is an attempt in quantifying scenarios wherein a breaking might occur and propose a generalized strategy to handle these situations.
+
+## Hard fork scenarios
+1. Change in any of the transaction specs
+    - Change in signature algorithm
+    - Change in hashing functions
+    - Change in schema
+2. Change in storage schema
+
+## Handling Hard Forks
+
+### Type 1 scenarios
+In case of change to spec it might not be possible to port all existing validated transaction to the new format. In such cases one needs to make a distinction between transactions which have already been validated and stored in the Blockchain from new incoming transactions. Furthermore, a suitable trigger is needed so that the network can collectively jump to the new logic else there could be situations wherein the network might not be able to commit any new blocks because of honest nodes running different validation logic. So, it is apparent that a mechanism for transitioning to this new validation logic is need. Below is one such procedure to achieve the same,
+
+- In order to decide when the validation logic should be upgraded to its newer version a TEP is initiated which defines the future block `height` at which to upgrade and the corresponding `BigchainDB` class to use.
+
+   NOTE: It is assumed that the new validation logic is implemented by extending the existing `BigchainDB` class and at the time of starting BigchainDB an instance of each of these classes is passed to the ABCI `App` i.e.,
+   ```python
+   validation = {
+       "v1": {
+           "start_block_height": 0,
+           "validation_class": "BigchainDB",
+       }
+       "v2": {
```

Comment:
Over time, the versions may accumulate so it might be a good idea to design a deprecation process offering a way to archive the old blocks and strip the old code off. We might not want to include it here to keep it simple though.

Reply:
This would be a tricky problem i.e. whenever a replay of all the blocks is required the old code will be needed. I am happy to explore possible these ideas further but that discussion would be out of the scope of this BEP.